src/db.c

```diff
@@ -765,31 +864,29 @@ void randomkeyCommand(client *c) {
 }
 
 void keysCommand(client *c) {
-    dictIterator *di;
     dictEntry *de;
     sds pattern = c->argv[1]->ptr;
     int plen = sdslen(pattern), allkeys;
-    unsigned long numkeys = 0;
+    long numkeys = 0;
     void *replylen = addReplyDeferredLen(c);
-
-    di = dictGetSafeIterator(c->db->dict);
+    dbIterator dbit;
+    dbIteratorInit(&dbit, c->db);
     allkeys = (pattern[0] == '*' && plen == 1);
-    while((de = dictNext(di)) != NULL) {
+    while ((de = dbIteratorNext(&dbit)) != NULL) {
         sds key = dictGetKey(de);
         robj *keyobj;
 
-        if (allkeys || stringmatchlen(pattern,plen,key,sdslen(key),0)) {
-            keyobj = createStringObject(key,sdslen(key));
-            if (!keyIsExpired(c->db,keyobj)) {
-                addReplyBulk(c,keyobj);
+        if (allkeys || stringmatchlen(pattern, plen, key, sdslen(key), 0)) {
+            keyobj = createStringObject(key, sdslen(key));
+            if (!keyIsExpired(c->db, keyobj)) {
+                addReplyBulk(c, keyobj);
                 numkeys++;
             }
```

Comment:
Nice! Now the indentation of these lines are not changed. Only some whitespace changes that can actually be reverted to get an even smaller diff.